File: Python/excel_sheet_column_title.py
# https://leetcode.com/problems/excel-sheet-column-title/description/
from utils.examination import Examiner, Testcase


class Solution:
    def convertToTitle(self, columnNumber: int) -> str:
        if 0 == columnNumber:
            return ''
        first_char: int = columnNumber % 26
        if 0 == first_char:
            first_char = 26
        return self.convertToTitle(int((columnNumber - first_char) / 26)) + chr(64 + first_char)

def examine() -> None:
    Examiner(Solution(), 'convertToTitle').run((
        Testcase((1,), 'A'),
        Testcase((28,), 'AB'),
        Testcase((701,), 'ZY'),
        Testcase((26,), 'Z'),
        Testcase((27,), 'AA'),
        Testcase((52,), 'AZ'),
    ))

File: TypeScript/solutions/excel-sheet-column-title.ts
// https://leetcode.com/problems/excel-sheet-column-title/description/
import { Examiner, Testcase } from '../utils';


function convertToTitle(columnNumber: number): string {
    if (columnNumber === 0) return '';
    const firstChar = columnNumber % 26 === 0 ? 26 : columnNumber % 26;
    return convertToTitle(Math.floor((columnNumber - firstChar) / 26)) + String.fromCharCode(64 + firstChar);
}

export function examine(): void {
    new Examiner(convertToTitle).run([
        new Testcase([1], 'A'),
        new Testcase([28], 'AB'),
        new Testcase([701], 'ZY'),
        new Testcase([26], 'Z'),
        new Testcase([27], 'AA'),
        new Testcase([52], 'AZ'),
    ]);
}
